tests/FakeItEasy.Specs/SelfInitializedFakesSpecs.cs

```diff
@@ -67,28 +209,181 @@ public static class SelfInitializedFakesSpecs
 
+        [Scenario]
+        public static void SelfInitializingThrowsIfTooManyCallsEncountered(
+            InMemoryStorage inMemoryStorage,
+            ILibraryService realServiceWhileRecording,
+            ILibraryService realServiceDuringPlayback,
+            Exception exception)
+        {
+            "Given a call storage object"
+                .x(() => inMemoryStorage = new InMemoryStorage());
+
+            "And a real service to wrap while recording"
+                .x(() => realServiceWhileRecording = A.Fake<ILibraryService>());
+
+            "And a real service to wrap while playing back"
+                .x(() => realServiceDuringPlayback = A.Fake<ILibraryService>());
+
+            "When I use a self-initialized fake in recording mode to get the count and title for book 1"
+                .x(() =>
+                {
+                    using (var recorder = new RecordingManager(inMemoryStorage))
+                    {
+                        var fakeService = A.Fake<ILibraryService>(options => options
+                            .Wrapping(realServiceWhileRecording).RecordedBy(recorder));
+
+                        fakeService.GetCount("1");
+                        fakeService.GetTitle("1");
+                    }
+                });
+
+            "And I use a self-initialized fake in playback mode to get the count and title and count for book 1"
+                .x(() =>
+                {
+                    using (var recorder = new RecordingManager(inMemoryStorage))
+                    {
+                        var playbackFakeService = A.Fake<ILibraryService>(options => options
+                            .Wrapping(realServiceDuringPlayback).RecordedBy(recorder));
+
+                        playbackFakeService.GetCount("1");
+                        playbackFakeService.GetTitle("1");
+                        exception = Record.Exception(() => playbackFakeService.GetCount("1"));
+                    }
+                });
+
+            // This result demonstrates that the self-initialized fake relies on a script
+            // defined by which methods are called, and is completely inflexible with
+            // regard to the order of calls.
+            "Then the playback fake throws a recording exception"
+                .x(() => exception.Should().BeAnExceptionOfType<RecordingException>());
+        }
+
+        [Scenario]
+        public static void SelfInitializingExceptionWhileRecording(
+            InMemoryStorage inMemoryStorage,
+            ILibraryService realServiceWhileRecording,
+            ILibraryService realServiceDuringPlayback,
+            Exception originalException,
+            Exception recordingException,
+            Exception playbackException)
+        {
+            "Given a call storage object"
+                .x(() => inMemoryStorage = new InMemoryStorage());
+
+            "And an exception"
+                .x(() => originalException = new InvalidOperationException());

```

Comment:
This step seems a little out of place; the exception isn't really part of the preconditions. The variable could be initialized when the fake is configured, e.g. `.Throws(originalException = new InvalidOperationException())`